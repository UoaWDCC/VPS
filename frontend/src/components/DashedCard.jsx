import { Box } from "@material-ui/core";

import AddRoundedIcon from "@mui/icons-material/AddRounded";

/**
 * Component used to represent a card with a dashed border, used to indicate that a new card can be created.
 *
 * @component
 * @example
 * function onClick() {
 *   console.log("Clicked")
 * }
 *
 * return (
 *   <DashedCard onClick={onClick} />
 * )
 */
export default function DashedCard({ onClick }) {
  return (
    <div>
      <div style={{ position: "relative" }}>
        <Box
          height={160}
          onClick={onClick}
          sx={{
            background: "#f1f5f9",
            "&:hover": {
              background: "#fff",
            },
          }}
          className="cursor-pointer flex justify-center items-center overflow-hidden rounded-xl border-2 border-dashed border-slate-400 bg-slate-100"
        >
          <AddRoundedIcon
            className="text-slate-500"
            sx={{
              fontSize: "5rem",
            }}
          />
        </Box>
      </div>
    </div>
  );
}
