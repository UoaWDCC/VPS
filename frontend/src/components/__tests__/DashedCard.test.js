import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import DashedCard from "../DashedCard";
import TestRenderer from "react-test-renderer";
const { act } = TestRenderer;

let component;
test("Dashed Card component snapshot test", () => {
  act(() => {
    component = renderer.create(
      <BrowserRouter>
        <DashedCard />
      </BrowserRouter>
    );
  });

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
