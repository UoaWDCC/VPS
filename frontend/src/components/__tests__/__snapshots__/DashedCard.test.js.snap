// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashed Card component snapshot test 1`] = `
<div
  className="imageListItemWide"
>
  <div
    className="MuiBox-root MuiBox-root-1"
  />
  <p
    className="text"
  >
    Create New Scene
  </p>
</div>
`;
